tests/unit/assessment/test_aws.py

```diff
@@ -0,0 +1,331 @@
+                        "s3:DeleteObject",
+                        "s3:PutObjectAcl",
+                        "s3:GetBucketNotification",
+                        "s3:PutBucketNotification"
+                    ],
+                    "Resource": [
+                        "arn:aws:s3:::bucketA/*",
+                        "arn:aws:s3:::bucketB/*",
+                        "arn:aws:s3:::bucketC/*"
+                    ]
+                }
+            ]
+        },
+        "VersionId": "v1",
+        "IsDefaultVersion": true,
+        "CreateDate": "2024-01-01T00:00:00+00:00"
+    }
+}
+    """
+
+    def command_call(cmd: str):
+        if cmd.startswith("aws iam get-role-policy"):
+            return 0, get_role_policy_return, ""
+        elif cmd.startswith("aws iam get-policy "):
+            return 0, get_policy_return, ""
+        elif cmd.startswith("aws iam get-policy-version"):
+            return 0, get_policy_version_return, ""
+        else:
+            return -1, "", "Error"
+
+    aws = AWSResources("Fake_Profile", command_call)
+    role_policy = aws.get_role_policy("fake_role", policy_name="fake_policy")
+    assert role_policy == [
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucket1",
+        ),
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucket2",
+        ),
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucket3",
+        ),
+    ]
+
+    role_policy = aws.get_role_policy("fake_role", attached_policy_arn="arn:aws:iam::12345:policy/policy")
+    assert role_policy == [
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucketA",
+        ),
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucketB",
+        ),
+        AWSPolicyAction(
+            resource_type="s3",
+            action={"s3:GetObject", "s3:PutObject", "s3:DeleteObject", "s3:PutObjectAcl"},
+            resource_path="bucketC",
+        ),
+    ]
+
+
+def test_save_instance_profile_permissions():
+    ws = create_autospec(WorkspaceClient)
+    ws.instance_profiles.list.return_value = [
+        InstanceProfile("arn:aws:iam::12345:instance-profile/role1", "arn:aws:iam::12345:role/role1")
+    ]
+    ws.current_user.me = lambda: iam.User(user_name="me@example.com", groups=[iam.ComplexValue(display="admins")])
+
+    expected_csv_entries = [
+        "arn:aws:iam::12345:instance-profile/role1,s3",
```

Comment:
In CSV files, you have to map if a role can read or write on a bucket (or prefix within a bucket). See data model in azure.py, make a similar one

Reply:
Done